### PR TITLE
proto: replace Borrow<Name> impl for LowerName with Deref

### PR DESCRIPTION
**Pull request by djc**
This is simpler to use and seems more idiomatic to me.

cc @pvdrz

### CODE REVIEW

**Review by marcus0x62 — APPROVED** [review]
This looks good to me.

**Comment by djc** [comment]
Let's wait for feedback from @pvdrz, feel free to merge once they approve.

**Pull request merged**
